src/main/java/seedu/algobase/model/problem/TagIncludesKeywordsPredicate.java

```diff
@@ -0,0 +1,40 @@
+package seedu.algobase.model.problem;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.algobase.model.tag.Tag;
+
+/**
+ * Tests that the given set of keywords is a subset of a {@code Problem}'s set of {@code Tag}.
+ */
+public class TagIncludesKeywordsPredicate implements Predicate<Problem> {
+
+    public static final TagIncludesKeywordsPredicate DEFAULT_TAG_PREDICATE =
+        new TagIncludesKeywordsPredicate() {
+            @Override
+            public boolean test(Problem problem) {
+                return true;
+            }
+        };
+    private final List<String> keywords;
```

Comment:
Can potentially abstract keywords out to a separate class

Reply:
Will do so if we require more complicated behavior from keywords.